src/modules/assessments/components/AssessmentForm.tsx

```diff
@@ -124,6 +125,13 @@ const AssessmentForm = ({
       },
     });
 
+  const handleDirty = useCallback(
+    (value: boolean) => {
+      onInflight?.(clientId, value);
+    },
+    [onInflight, clientId]
+  );
+
```

Comment:
Oh yeah, I started by naming it in-flight thinking it just needed to track requests that had not returned but then discovered what I really needed to track was if the form was dirty, otherwise you end up with weird UI flickering on the submit tab since it renders and then fires the mutation. I'll make a pass back through and rename inFlight to dirty, thanks for calling that out.

I agree, I think pulling state and actions up out of the form seems like a good direction when we get a chance to refactor.